src/plugins/intel_gpu/tests/unit/transformations/lora_horizontal_fusion.cpp

```diff
@@ -0,0 +1,317 @@
+// Copyright (C) 2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <gtest/gtest.h>
+
+#include "common_test_utils/ov_test_utils.hpp"
+#include "openvino/opsets/opset6.hpp"
+
+#include "plugin/transformations/lora_horizontal_fusion.hpp"
+#include "intel_gpu/op/placeholder.hpp"
+#include "intel_gpu/op/fully_connected_compressed.hpp"
+
+using namespace testing;
+using namespace ov::intel_gpu;
+
+namespace ov {
+namespace test {
+namespace intel_gpu {
+
+TEST_F(TransformationTestsF, LoRAHorizontalFusion_default) {
+    ov::element::Type model_dt = ov::element::f16;
+    {
+        auto lora_input = std::make_shared<ov::op::v0::Parameter>(model_dt, ov::PartialShape{-1, -1, 2048});
+        auto weights = std::make_shared<ov::op::v0::Constant>(ov::element::u8, ov::Shape{2560, 2048});
+        auto bias = std::make_shared<ov::intel_gpu::op::Placeholder>();
+        auto scale = std::make_shared<ov::op::v0::Constant>(model_dt, ov::Shape{2560, 1});
+        auto fc_fused = std::make_shared<ov::intel_gpu::op::FullyConnectedCompressed>(lora_input, weights, bias, scale);
+
+        auto axis_const = ov::op::v0::Constant::create(ov::element::i64, ov::Shape{1}, {2});
+        auto split_const = ov::op::v0::Constant::create(ov::element::i64, ov::Shape{3}, {2048, 256, 256});
+        auto split = std::make_shared<ov::op::v1::VariadicSplit>(fc_fused, axis_const, split_const);
```

Comment:
Will the pass work if the Split op divides the tensor into 2 or 4 parts? I don't think that you check that somewhere in the transformation.
Another thing is multiple users of Split outputs - please check that too.